src/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -0,0 +1,145 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    internal class AuthorityInfo
+    {
+        public AuthorityInfo(
+            AuthorityType authorityType,
+            string authority,
+            bool isDefault)
+        {
+            AuthorityType = authorityType;
+            IsDefault = isDefault;
+
+            Host = new UriBuilder(authority).Host;
+
+            // TODO: can we simplify this and/or move validation/configuration logic to AbstractApplicationBuilder
+            // so that all authority mangling/management is in one place?
+
+            UserRealmUriPrefix = string.Format(CultureInfo.InvariantCulture, "https://{0}/common/userrealm/", Host);
+
+            if (AuthorityType == AuthorityType.B2C)
+            {
+                var authorityUri = new Uri(authority);
```

Comment:
seems to me that some of this functionality belongs to the Authority classes and not here?

Reply:
If you look through the whole "markzuber/config" branch you'll see that the Authority class will own the AuthorityInfo class, do some work, and then spit out an AuthorityEndpoints class that has all of the goodies that the rest of the flows need.  This is work on the backlog I sent out to the team.

---
In reply to: [246489799](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/761#discussion_r246489799) [](ancestors = 246489799)